build-system/tasks/amp-task-runner.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This is a lightweight task runner with a one-file implentation
+ * based on the commander npm package. It behaves like gulp without the overhead
+ * of a large set of dependencies.
+ */
+
+const argv = require('minimist')(process.argv.slice(2));
+const commander = require('commander');
+const os = require('os');
+const path = require('path');
+const {cyan, red, green, magenta} = require('kleur/colors');
+const {log} = require('../common/logging');
+
+/**
+ * Special-case constant that indicates if `amp --help` was invoked.
+ */
+const isHelpTask = argv._.length == 0 && argv.hasOwnProperty('help');
+
+/**
+ * Calculates and formats the duration for which an AMP task ran.
+ * @param {DOMHighResTimeStamp} start
+ * @return {string}
+ */
+function getTime(start) {
+  const endTime = Date.now();
+  const executionTime = endTime - start;
+  const mins = Math.floor(executionTime / 60000);
+  const secs = Math.floor((executionTime % 60000) / 1000);
+  const msecs = executionTime % 1000;
+  return mins != 0
+    ? `${mins}m ${secs}s`
+    : secs != 0
+    ? `${secs}s`
+    : `${msecs}ms`;
+}
+
+/**
+ * Runs an AMP task with logging and timing.
+ * @param {string} taskName
+ * @param {Function()} taskFunc
+ * @return {Promise<void>}
+ */
+async function runTask(taskName, taskFunc) {
+  const taskFile = '~' + path.sep + path.relative(os.homedir(), 'amp.js');
```

Comment:
`~` to represent the user's home directory is a POSIX concept, so this might not be familiar to Windows users. It's also possible that users will place the files outside of their homedir, which would cause something odd to appear such as `~/some/other/directory/../../../build-system/tasks/amp.js`

Reply:
I did this to closely match the [original behavior](https://github.com/gulpjs/gulp-cli/blob/da8241ecbacd59158deaa5471ff8a7f43901a94b/lib/shared/tildify.js#L5-L7). Fair point though. I've simplified this by just using `amp.js`. But we do need to start all tasks at the repo root (in case someone tries running a task from a subdirectory.)